clients/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractSparkSqlTest.java

```diff
@@ -18,140 +18,30 @@
-
-  private Branch initialDefaultBranch;
-
-  private String refName;
-  private String additionalRefName;
-  protected NessieApiV1 api;
-
-  @BeforeEach
-  void setupSparkAndApi(TestInfo testInfo) throws NessieNotFoundException {
-    api = HttpClientBuilder.builder().withUri(url).build(NessieApiV1.class);
-
-    refName = testInfo.getTestMethod().map(Method::getName).get();
-    additionalRefName = refName + "_other";
-
-    initialDefaultBranch = api.getDefaultBranch();
-
-    Map<String, String> nessieParams =
-        ImmutableMap.of(
-            "ref", defaultBranch(), "uri", url, "warehouse", tempFile.toURI().toString());
-
-    nessieParams.forEach(
-        (k, v) -> {
-          conf.set(String.format("spark.sql.catalog.nessie.%s", k), v);
-          conf.set(String.format("spark.sql.catalog.spark_catalog.%s", k), v);
-        });
-
-    conf.set(SQLConf.PARTITION_OVERWRITE_MODE().key(), "dynamic")
-        .set("spark.testing", "true")
-        .set("spark.sql.warehouse.dir", tempFile.toURI().toString())
-        .set("spark.sql.shuffle.partitions", "4")
-        .set("spark.sql.catalog.nessie.catalog-impl", "org.apache.iceberg.nessie.NessieCatalog")
-        .set("spark.sql.catalog.nessie", "org.apache.iceberg.spark.SparkCatalog");
-
-    // the following catalog is only added to test a check in the nessie spark extensions
-    conf.set(
-            String.format("spark.sql.catalog.%s", NON_NESSIE_CATALOG),
-            "org.apache.iceberg.spark.SparkCatalog")
-        .set(
-            String.format("spark.sql.catalog.%s.catalog-impl", NON_NESSIE_CATALOG),
-            "org.apache.iceberg.hive.HiveCatalog");
-
-    spark = SparkSession.builder().master("local[2]").config(conf).getOrCreate();
-    spark.sparkContext().setLogLevel("WARN");
-  }
-
-  @AfterEach
-  void removeBranches() throws NessieConflictException, NessieNotFoundException {
-    // Reset potential "USE REFERENCE" statements from previous tests
-    SparkSession.active()
-        .sparkContext()
-        .conf()
-        .set(String.format("spark.sql.catalog.%s.ref", "nessie"), defaultBranch())
-        .remove(String.format("spark.sql.catalog.%s.ref.hash", "nessie"));
-
-    Branch defaultBranch = api.getDefaultBranch();
-    for (Reference ref : api.getAllReferences().get().getReferences()) {
-      if (ref instanceof Branch && !ref.getName().equals(defaultBranch.getName())) {
-        api.deleteBranch().branchName(ref.getName()).hash(ref.getHash()).delete();
-      }
-      if (ref instanceof Tag) {
-        api.deleteTag().tagName(ref.getName()).hash(ref.getHash()).delete();
-      }
-    }
-    api.assignBranch().assignTo(initialDefaultBranch).branch(defaultBranch).assign();
-    api.close();
-    api = null;
-  }
-
-  @AfterAll
-  static void tearDown() {
-    if (spark != null) {
-      spark.stop();
-      spark = null;
-    }
-  }
+public abstract class AbstractSparkSqlTest extends SparkSqlTestBase {
```

Comment:
nit: maybe rename this to `AbstractNessieSparkSqlExtentionTest` for clarity? Otherwise, its purpose is not very apparent when compared to `SparkSqlTestBase`. WDYT?